ui/src/app/connect/dialog/start-all-adapters/start-all-adapters-dialog.component.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { DialogRef } from '@streampipes/shared-ui';
+import {
+    AdapterDescriptionUnion,
+    AdapterService,
+} from '@streampipes/platform-services';
+
+@Component({
+    selector: 'sp-start-all-adapters-dialog',
+    templateUrl: './start-all-adapters-dialog.component.html',
+    styleUrls: ['./start-all-adapters-dialog.component.scss'],
+})
+export class StartAllAdaptersDialogComponent implements OnInit {
+    @Input()
+    adapters: AdapterDescriptionUnion[];
+
+    adaptersToModify: AdapterDescriptionUnion[];
+    installationStatus: any;
+    installationFinished: boolean;
+    page: string;
+    nextButton: string;
+    installationRunning: boolean;
+
+    @Input()
+    action: boolean;
+
+    constructor(
+        private dialogRef: DialogRef<StartAllAdaptersDialogComponent>,
+        private adapterService: AdapterService,
+    ) {
+        this.adaptersToModify = [];
+        this.installationStatus = [];
+        this.installationFinished = false;
+        this.page = 'preview';
+        this.nextButton = 'Next';
+        this.installationRunning = false;
+    }
+
+    ngOnInit() {
+        this.getAdaptersToModify();
+        if (this.adaptersToModify.length === 0) {
+            this.nextButton = 'Close';
+            this.page = 'installation';
+        }
+    }
+
+    close(refreshAdapters: boolean) {
+        this.dialogRef.close(refreshAdapters);
+    }
+
+    next() {
+        if (this.page === 'installation') {
+            this.close(true);
+        } else {
+            this.page = 'installation';
+            this.initiateInstallation(this.adaptersToModify[0], 0);
+        }
+    }
+
+    getAdaptersToModify() {
+        this.adapters.forEach(adapter => {
+            this.adaptersToModify.push(adapter);
```

Comment:
Why do we need the `adaptersToModify` array even though always all adapters are added to the array?
Do we need to check for the current status to make sure that the request is not executed for already started/stopped adapters?

Reply:
I initially added a condition to check the running status of the current adapter to execute the request only for started/stopped adapters but I was unable to figure out how to access the `running` field. So I commented out that condition and this is what I was left with...

Reply:
Can this be circumvented by filtering the list of adapters for `AdapterStreamDescription` and the for the running field so that sets are omitted?

Reply:
Will try this and let you know if I face any issues!

Reply:
Filtering the list for `AdapterStreamDescription` worked!